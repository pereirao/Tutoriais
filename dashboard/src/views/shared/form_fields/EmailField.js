// @flow

import React, {PureComponent} from 'react';
import _ from 'lodash';
import I18n from '../../../localization';
import style_sheet from './form_fields.module.css';

const email_error_text = {
  required: I18n.t('form.error.required_field')
};
const email_validators = {
  required: val => !!val
};

type EmailFieldProps = { onChangeText: (string, Object) => void, is_required: boolean };

class EmailField extends PureComponent {
  props: EmailFieldProps

  static defaultProps = {name: 'email', is_required: true}
  input: ReactClass<*>

  getValidity = value => (
    this.props.is_required ? _.mapValues(email_validators, validate => validate(value)) : true
  )

  onChangeText = value => {
    this.props.onChangeText(value, this.getValidity(value));
  }

  render() {
    return (
      <input
        errorText={email_error_text}
        placeholder={I18n.t(`form.placeholder.${this.props.name}`)}
        ref={el => {
          this.input = el;
        }}
        className={style_sheet.textInput}
        {...this.props}
        onChange={({target}) => this.onChangeText(target.value)}/>
    );
  }
}

export default EmailField;
