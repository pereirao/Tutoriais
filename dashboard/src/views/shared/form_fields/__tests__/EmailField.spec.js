import React from 'react';

import EmailField from '../EmailField';

it('renders correctly', () => {
  expect(shallow(<EmailField />)).toMatchSnapshot();
});
