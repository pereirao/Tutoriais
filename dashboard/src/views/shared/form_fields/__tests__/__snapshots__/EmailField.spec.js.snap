// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  className="textInput"
  errorText={
    Object {
      "required": "This field is required",
    }
  }
  is_required={true}
  name="email"
  onChange={[Function]}
  placeholder="Email Address"
/>
`;
